simulator/modules/branch/branch.h

```diff
@@ -45,6 +45,22 @@ class Branch : public Log
         auto get_mispredict_rate() const { return ( num_branches != 0) 
                                                   ? num_mispredictions / num_branches * 100
                                                   : 0; }
+        template<typename Instr>
+        bool is_misprediction( const Instr& instr, const BPInterface& bp_data)
+        {
+            bool is_misprediction = false;
+
+            if ( instr.is_indirect_jump())
+                is_misprediction |= bp_data.target != instr.get_new_PC();
```

Comment:
You can just ```return bp_data.target != instr.get_new_PC()``` here

Reply:
As I understand, at this exact moment 'bp_data' keeps information about BPU result and 'Instr' already contains the actual information. That's why I've made is_misprediction() more complicated. Could you please check it?

Reply:
Nope, that does not look correct at all. On branch stage, we need to compare BP data with the actual data.

Reply:
Ok, I will fix that. But firstly, I want to make sure that I understand everything right.

- 'bp_data' contains information about prediction ( was 'jump' taken or not, what was predicted 'target').
- 'Instr' contains actual information ( should we take this 'jump' or not, what is the actual 'target' for it).
Is this correct?

Reply:
Yes. For direct jumps and predicted-to-taken branches we compare target at decode stage, everything else is compared on branch resolution stage.

Reply:
Thanks, that makes sense.